### DISCUSSION TITLE
Wochenprogramm über Dienst wechseln

### DISCUSSION THREAD

**Discussion by lgusewski** [SukramJ/aiohomematic#1320] [Q&A] [answered]
Hallo, ich habe die Version 1.50 der custom_homematic Integration unter HA 2023.12.1 installiert. Für meine Räume gibt es in der RaspberryMatic (3.73.9.20231130) jeweils eine Gruppe. Diese sind in HA angelegt, die einzelnen Thermostate habe ich ausgeblendet. Nun wollte ich eine möglichst einfache Automatisation schreiben, die nach dem integrierten Kalender in den Schulferien das Wochenprogramm von 1 auf 2 ändert. In der Oberfläche werden die Programme von HA als "WP 1", "WP 2" und "WP 3" aufgelistet. Die Änderung über den Dienst "climate.set_preset_mode" klappt aber weder über "2" oder "WP 2". Es kommt z.B. eine Fehlermeldung "Preset mode 2 is not supported in hvac_mode auto".
Ich habe zwar eine Lösung gefunden, die an das Gruppen-Gerät auf den Kanal 1 den Parameter "ACTIVE_PROFILE" auf 2 ändert, aber das ist wesentlich unübersichtlicher, vor allem, wenn man mehrere Gruppen ändern möchte. Zudem wäre ich ohne Suche im Netz nie selber darauf gekommen. Geht das Umschalten der Wochenprogramme überhaupt über den "climate.set_preset_mode"-Dienst?

**Comment by SukramJ** [comment, marked as answer]
WP 1 ist nur eine „Übersetzung“.

Der zu verwendende preset mode ist week_program_1.

Das kann man in den Entwicklerwerkzeugen unter Zuständen an den jeweiligen Entitäten sehen.

**Comment by lgusewski** [comment]
Danke für diese wahnsinnig schnelle Antwort. Es hat geklappt. Wie man vielleicht erahnen kann, bin ich noch ein HA Neuling. Durch Deinen Kommentar habe ich allerdings auch die Stelle in den Entwicklerwerkzeugen finden können, in denen die Einträge für die gültigen Namen der Wochenprogramm aufgelistet werden und die Automatisation bereits vereinfacht.

**Reply by SukramJ** [reply]
Gerne. Über die Ui kannst du das Ganze auch zusammenklicken.
Als Action das Gerät auswählen …
